spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/convert/MongoConverter.java

```diff
@@ -149,4 +149,15 @@ default Object convertId(@Nullable Object id, Class<?> targetType) {
 			return convertToMongoType(id, null);
 		}
 	}
+
+	/**
+	 * Compute the target type for a given source considering {@link org.springframework.data.convert.CustomConversions}.
+	 *
+	 * @param source the source type.
+	 * @return never {@literal null}.
+	 * @since 2.2
+	 */
+	default Class<?> computeWriteTarget(Class<?> source) {
```

Comment:
We should improve either on the definition or on the implementation of this method. Right now, we could assume that the write target is either a native Mongo type or the write target is an to be converted type.

The number of exposed methods via `MongoWriter` and `MongoConverter` keeps constantly increasing and I'm not sure how to think about it.